### PR TITLE
Global search service

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-6230

## Changes
- Add GlobalSearchService interface and service configuration support
- Implement FuiSearchService

### CODE REVIEW

File: src/foam/nanos/boot/NSpec.js [hunk 1]
```diff
@@ -215,6 +215,17 @@ foam.CLASS({
       transient: true,
       javaGetter: 'return getName();',
       getter: function() { return this.name; }
+    },
+    {
+      class: 'Int',
+      name: 'uidKey',
+      value: -1,
+      javaSetter: `
+        uidKey_ = val;
```
Comment (jlhughes) [inline src/foam/nanos/boot/NSpec.js]: curious why this is needed?

Comment (lchanmann) [reply src/foam/nanos/boot/NSpec.js]: The uidKey will be set in easyDAO when setting up the FUIDDAO, but the nspec instance in the easyDAO is frozen and I'm hesitant to fclone the nspec to update the uidKey then save back to the nspecDAO.

Might be a bit hacky this way, I'm open to suggestions @jlhughes @kgrgreer.

Comment (lchanmann) [reply src/foam/nanos/boot/NSpec.js]: Removed 'uidKey' and use fuid keys instead.

**Review by jlhughes — COMMENTED** [review]
@lchanmann I hate to ask, but I would like to change the package.
when I think of foam.util.search I imagine general search support of some kind, but this is specific to foam's uid.  How about foam.util.uid. ?

**Review by jlhughes — APPROVED** [review]

**Pull request merged**